keras/src/wrappers/random_state.py

```diff
@@ -0,0 +1,73 @@
+import os
+import random
+from contextlib import contextmanager
+from typing import Generator
+from typing import Iterator
+
+import numpy as np
+
+try:
+    import tensorflow as tf
+    from tensorflow.python.eager import context
+    from tensorflow.python.framework import config
+    from tensorflow.python.framework import ops
+
+    def tf_set_seed(seed: int) -> None:
+        tf.random.set_seed(seed)
+
+    def tf_get_seed() -> Iterator[int]:
+        if context.executing_eagerly():
+            return context.global_seed()
+        else:
+            return ops.get_default_graph().seed
+
+    def tf_enable_op_determinism() -> bool:
+        was_enabled = config.is_op_determinism_enabled()
+        config.enable_op_determinism()
+        return was_enabled
+
+    def tf_disable_op_determinism() -> None:
+        config.disable_op_determinism()
+
+except ImportError:
+
+    def tf_set_seed(seed: int) -> None:
+        pass
+
+    def tf_get_seed() -> int:
+        return 0
+
+    def tf_enable_op_determinism() -> bool:
+        return False
+
+    def tf_disable_op_determinism() -> None:
+        return None
+
+
+@contextmanager
+def tensorflow_random_state(seed: int) -> Generator[None, None, None]:
```

Comment:
This should really not be necessary, instead just use `set_random_seed` from Keras. If users need TF_DETERMINISTIC_OPS that is something they should set manually, separately.